src/System/input_enabler.rs

```diff
@@ -125,49 +100,40 @@ pub fn make_input_enabled(
 }
 
 fn build_guard_from_zone(
-    zone: *const i32,
-    dimension: u32,
+    zone: &mut Zone,
     clocks: &HashMap<String, u32>,
 ) -> Option<representations::BoolExpression> {
     let mut guards: Vec<representations::BoolExpression> = vec![];
 
     for (_, index) in clocks {
-        let raw_upper = lib::rs_dbm_get_constraint_from_dbm_ptr(zone, dimension, *index, 0);
-        let raw_lower = lib::rs_dbm_get_constraint_from_dbm_ptr(zone, dimension, 0, *index);
+        let (raw_upper_rel, raw_upper_val) = zone.get_constraint(*index, 0);
+        let (raw_lower_rel, raw_lower_val) = zone.get_constraint(0, *index);
 
         // lower bound must be different from 1 (==0)
-        if raw_lower != 1 {
-            if lib::rs_raw_is_strict(raw_lower) {
+        if !(raw_lower_rel && raw_lower_val == 0) {
```

Comment:
pretty sure this should be !(!raw_lower_rel && raw_lower_val == 0).
The check is whether the constraint is (<=0), and raw_lower_rel is true if it is strict e.g. (<0)

Reply:
```suggestion
        if !(!raw_lower_rel && raw_lower_val == 0) {
```